custom/enikshay/management/commands/import_voucher_confirmations.py

```diff
@@ -0,0 +1,234 @@
+        'EnikshayApprover',
+        'EnikshayRole',
+        'EnikshayApprovalDate',
+    ]
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('filename')
+        parser.add_argument(
+            '--commit',
+            action='store_true',
+            dest='commit',
+            default=False,
+        )
+
+    def handle(self, domain, filename, **options):
+        self.domain = domain
+        self.accessor = CaseAccessors(domain)
+        self.commit = options['commit']
+
+        with open(filename) as f:
+            reader = csv.reader(f)
+            headers = reader.next()
+            missing_headers = set(self.voucher_update_properties) - set(headers)
+            if missing_headers:
+                print "Missing the following headers:"
+                for header in missing_headers:
+                    print " ", header
+                print "\nAborting."
+                return
+
+            rows = [r for r in reader]
+
+        print "Received info on {} vouchers.  Headers are:".format(len(rows) - 1)
+        for header in headers:
+            print header
+
+        voucher_updates = []
+        unrecognized_vouchers = []
+        voucher_ids_to_update = set()
+
+        for row in rows:
+            voucher_id = row[headers.index(self.voucher_id_header)]
+            voucher = self.all_vouchers_in_domain.get(voucher_id)
+            if voucher:
+                voucher_ids_to_update.add(voucher_id)
+                voucher_updates.append(VoucherUpdate(
+                    voucher=voucher,  # This property isn't defined on the model
+                    id=voucher.case_id,
+                    **{
+                        prop: row[headers.index(prop)]
+                        for prop in self.voucher_update_properties
+                    }
+                ))
+            else:
+                unrecognized_vouchers.append(row)
+
+        self.log_voucher_updates(voucher_updates)
+        self.log_unrecognized_vouchers(headers, unrecognized_vouchers)
+        self.log_unmodified_vouchers(voucher_ids_to_update)
+        self.update_vouchers(voucher_updates)
+        self.reconcile_repeat_records(voucher_updates)
+
+    @property
+    @memoized
+    def all_vouchers_in_domain(self):
+        voucher_ids = self.accessor.get_case_ids_in_domain(CASE_TYPE_VOUCHER)
+        return {
+            voucher.get_case_property(VOUCHER_ID): voucher
+            for voucher in self.accessor.iter_cases(voucher_ids)
+            if voucher.get_case_property('state') in (
+                'approved', 'paid', 'rejected', 'expired', 'cancelled', 'canceled')
+            or voucher.get_case_property('voucher_approval_status') in ('approved', 'partially_approved')
+        }
+
+    def write_csv(self, filename, headers, rows):
+        filename = "voucher_confirmations-{}.csv".format(filename)
+        print "writing {}".format(filename)
+        with open(filename, 'w') as f:
```

Comment:
I think that's fine - if a file already exists, it's because I'm running this same command multiple times.  I'm planning to do a no-commit test run, look at the results, then do a real run.  I'd expect to overwrite the files from the test run.